apps/web/src/components/SearchRequestTable/SearchRequestsTableApi.tsx

```diff
@@ -30,13 +35,48 @@ import useTableState from "~/components/Table/ApiManagedTable/useTableState";
 import { tableViewItemButtonAccessor } from "~/components/Table/ClientManagedTable/TableViewItemButton";
 import useRoutes from "~/hooks/useRoutes";
 import {
-  getLocalizedName,
-  getPoolCandidateSearchStatus,
-} from "@gc-digital-talent/i18n";
-import { formatDate, parseDateTimeUtc } from "@gc-digital-talent/date-helpers";
+  stringToEnumRequestStatus,
+  stringToEnumStream,
+} from "~/utils/requestUtils";
+
+import SearchRequestsTableFilter, {
+  FormValues,
+} from "./components/SearchRequestsTableFilterDialog";
 
 type Data = NonNullable<FromArray<PoolCandidateSearchRequestPaginator["data"]>>;
 
+// data massaging functions
+function transformFormValuesToSearchRequestFilterInput(
+  data: FormValues,
+): PoolCandidateSearchRequestInput {
+  return {
+    status: data.status.map((status) => {
+      return stringToEnumRequestStatus(status);
+    }),
+    departments: data.departments,
+    classifications: data.classifications,
+    streams: data.streams.map((stream) => {
+      return stringToEnumStream(stream);
+    }),
```

Comment:
If you like succinct code...
```suggestion
    status: data.status.map(stringToEnumRequestStatus),
    departments: data.departments,
    classifications: data.classifications,
    streams: data.streams.map(stringToEnumStream),
```

Reply:
Sure! https://github.com/GCTC-NTGC/gc-digital-talent/pull/6897/commits/b310b2318cff0894762c592b54328556c68e501e